### DISCUSSION TITLE
Insert blocks from template file

### DISCUSSION THREAD

**Discussion by mindreaderlupo** [tarikjabiri/dxf#131] [Q&A]
I have very complicated blocks with a lot of shapes in a template DXF file. I want to insert them and manipulate the position but without reverse engineering the content of each block.
Is there any possibility or workaround for that?

Thank You!

**Comment by tarikjabiri** [comment]
Hi

I think you need to parse the template file and extract those blocks, so you can use/insert them later
You can use https://github.com/dxfjs/parser as parser to extract the blocks

Regards

**Reply by mindreaderlupo** [reply]
Thanks! For simple DXF files this works perfect, for more complex I have issues because the Parser uncovers errors in the DXF which I can not fix. But this is an issue with my file quality.

Is there a possibility to add properties to blocks to keep them dynamic and manipulate them according to rules in AutoCAD?

Thanks 
Felix

**Reply by tarikjabiri** [reply]
Hi, I didnt get what you mean by properties to keep them dynamic 🤔.
Can you link a dxf not working, to see if I can understand what happening.

**Reply by mindreaderlupo** [reply]
Ok, I described it in a strage way :) What I actually mean is dynamic blocks:
https://www.autodesk.com/blogs/autocad/how-to-create-a-dynamic-block-and-save-time/

Here is an example file, containing dynamic Blocks:
[https://github.com/mindreaderlupo/DXF-Parse/blob/main/src/assets/ExampleDynamicBlock.dxf]

**Reply by tarikjabiri** [reply]
Ok I see, unfortunately this is not supported, I dont know how those properties are stored in dxf.
